### ISSUE TITLE
Getting "null" and "Debug adapter process has terminated unexpectedly" when debugging

### ISSUE THREAD

**Issue by AngrySaltyFish** [microsoft/vscode-cpptools#313]
Hello,

I am currently having a issue where I am unable to run a executable compile with g++ and get the following errors:
![deepinscreenshot20161027142809](https://cloud.githubusercontent.com/assets/16545605/19772280/40a18bcc-9c55-11e6-92dc-a7c25b8e8e4a.png)

My "launch.json":

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "C++ Launch",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceRoot}/a.out",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceRoot}",
            "environment": [],
            "externalConsole": true,
            "linux": {
                "MIMode": "gdb"
            },
            "osx": {
                "MIMode": "lldb"
            },
            "windows": {
                "MIMode": "gdb"
            }
        },
        {
            "name": "C++ Attach",
            "type": "cppdbg",
            "request": "attach",
            "program": "${workspaceRoot}/a.out",
            "processId": "${command.pickProcess}",
            "linux": {
                "MIMode": "gdb"
            },
            "osx": {
                "MIMode": "lldb"
            },
            "windows": {
                "MIMode": "gdb"
            }
        }
    ]
}
```

gdb Seems to be working fine in a normal shell and i am currently running arch linux.

**Comment by pieandcakes** [comment]
@Archangel7777777 Can you provide which version of OS you are using and what version of the extension?

**Comment by AngrySaltyFish** [comment]
@pieandcakes 
Os version is `4.8.4-1-ARCH`
extension version is `0.9.2`

**Comment by pieandcakes** [comment]
@Archangel7777777 Can you tell me what you are doing when it gives you the exception? Specifically some steps where I can reproduce this issue?

Also, can you enable **trace** and **traceResponse** and **engineLogging** in the `launch.json` and send me what you get when this error occurs?

Ex:
     `"logging": {
                "trace": true,
                "traceResponse": true,
                "engineLogging": true
            }`

**Comment by AngrySaltyFish** [comment]
@pieandcakes 
Apologies for not giving much detail so far, i have bit busy recently so hopefully this makes up for it. 

In terms steps to reproduce I don't really have any. Originaly I only had the `Debug adapter process has terminated unexpectedly` when running the launch config. After doing some googling i found #100 which said that after upgrading his system resolved the problem. After upgrading my system nothing changed so in desperate effort I uninstalled the [visual-studio-code](https://aur.archlinux.org/packages/visual-studio-code/) and installed [visual-studio-code-git](https://aur.archlinux.org/packages/visual-studio-code-git/). Finally after installling the extentsion again I get `null` and `Debug adapter process has terminated unexpectedly`. 

In terms of the logging I added what you suggest to the launch config and ran it. Nothing appeared the debug console (am I looking in the right place?). 

I did try turning on developer tools and had a look in the console and found the following:
![deepinscreenshot20161101164732](https://cloud.githubusercontent.com/assets/16545605/19898225/086a118c-a053-11e6-9c49-c4c2b91605f3.png)

Not sure if any helps as I new to visual code studio so I have no idea of what I am doing let me know if you need anything else.

**Comment by pieandcakes** [comment]
@Archangel7777777 Can you try reinstalling the extension? you can do that in the extension manager and search for the C++ extension and choose the uninstall option and then reinstall it again.

The logging I had you enable should show up in the Debugger Console. What that tells me is that there is something wrong with the installation. 

The other item it might be is that it can't find gdb. you can add the tag `"miDebuggerPath": "location/to/your/gdb"` to  your launch.json and see if that helps.

**Comment by AngrySaltyFish** [comment]
@pieandcakes I managed to reinstall the extenstion. According to the debug console all dependicies were installed correctly and it was finised installing. After reloading I tried debugging but found that it hung and kept loading even after pressing the red stop button. Eventaully I exited vs code and found the same error as before when trying to debug.

**Comment by AngrySaltyFish** [comment]
@pieandcakes I did also try using the tag suggested unfortunatly nothing changed!

**Comment by pieandcakes** [comment]
@Archangel7777777 Can you see if you have mono.linux running? You can do `ps aux | grep mono` in a terminal window. it sounds like the first time you ran the debugger, you are saying that it "hung" ? can you provide more information or a screenshot?

**Comment by AngrySaltyFish** [comment]
@pieandcakes Good news it's Fixed! It turns out there was a problem with mono.

After finding out that `mono.linux` was not running I thought it would be a good idea to check whether mono was working. After running monodevelop from the terminal I got `System.InvalidTimeZoneException: TZ file too big for the buffer`. This suggested a problem with the time zone so I ran the following commands to try and fix it:

`rm /etc/localtime`
`ln -s /usr/share/zoneinfo/`

After that monodevelop was up and running and vs code was debugging.

Thank you very much for your help even though it was not you problem in the end.
